server/lib/chat/chat-service.ts

```diff
@@ -909,21 +910,24 @@ export default class ChatService {
       return
     }
 
-    const channelSet = Set(userChannels.map(c => c.channelId))
-    const userSet = Set<SbUserId>(userChannels.map(u => u.userId))
-    const inChannels = Map(userChannels.map(c => [c.channelId, userSet]))
+    const channelIdsSet = Set(userChannels.map(c => c.channelId))
+    const userIdsSet = Set(userChannels.map(u => u.userId))
+    const inChannels = Map(userChannels.map(c => [c.channelId, userIdsSet]))
 
     this.state = this.state
       .mergeDeepIn(['channels'], inChannels)
-      .setIn(['users', userSockets.userId], channelSet)
+      .setIn(['users', userSockets.userId], channelIdsSet)
     for (const userChannel of userChannels) {
       this.publisher.publish(getChannelPath(userChannel.channelId), {
         action: 'userActive2',
         userId: userSockets.userId,
       })
       this.subscribeUserToChannel(userSockets, userChannel.channelId)
     }
-    userSockets.subscribe(`${userSockets.getPath()}/chat`, () => ({ type: 'chatReady' }))
+    userSockets.subscribe<ChatReadyEvent>(`${userSockets.getPath()}/chat`, () => ({
+      type: 'chatReady',
+      channelIds: channelIdsSet.toArray(),
```

Comment:
This will always use the channel IDs that were set when the user connected, so it is broken if they join or leave a channel and then connect a new socket. You need to re-retrieve it from the state or DB.

Whenever you write a `subscribe` data getter, you should almost never be using values from the closure. That is very rarely a correct thing to do (if it were correct, you could simply pass the object and not use a getter function)